contracts/contracts/pausing/Pauser.sol

```diff
@@ -0,0 +1,232 @@
+    event StrategistChanged(address _newStrategist);
+    event TempPaused(uint256 _expiry);
+    event Paused();
+    event Unpaused();
+    event Whitelisted(address _account);
+    event DeWhitelisted(address _account);
+
+    uint8 public constant UNPAUSED = 0;
+    uint8 public constant TEMP_PAUSE = 1;
+    uint8 public constant PAUSED = 2;
+
+    /**
+     * @dev The current state of the contract.
+     */
+    uint8 public pauseState;
+
+    /**
+     * @dev The duration of a temp-pause in seconds.
+     */
+    uint256 public expiryDuration;
+
+    /**
+     * @dev The expiry timestamp of the current temp-pause.
+     * If unpaused or confirmed, the expiry timestamp should be zero (0).
+     */
+    uint256 public currentExpiry;
+
+    /**
+     * @dev The address of the managed contract.
+     */
+    address public pausable;
+
+    /**
+     * @dev Address of the strategist
+     */
+    address public strategistAddr = address(0);
+
+    /**
+     * @dev whitelist of addresses allowed to interact with the functionality exposed by this contract.
+     */
+    mapping(address => bool) public whitelist;
+
+    /**
+     * @dev Expiry of the pause initiated by the account.
+     * Map of account to the expiry of the temp-pause they have initiated.
+     * Also used to invalidate a pauser to prevent abuse.
+     */
+    mapping(address => uint256) public pauseExpiry;
+
+    /**
+     * @dev Verifies that the caller is a whitelisted account or the governor or strategist.
+     */
+    modifier onlyWhitelistedOrGovernorOrStrategist() {
+        require(
+            whitelist[msg.sender] || _isGovernorOrStrategist(),
+            "Caller is not whitelisted and is not the Strategist or Governor"
+        );
+        _;
+    }
+
+    modifier onlyGovernorOrStrategist() {
+        require(
+            _isGovernorOrStrategist(),
+            "Caller is not the Strategist or Governor"
+        );
+        _;
+    }
+
+    function initialize(address _pausable, uint256 _expiryDuration)
+        external
+        onlyGovernor
+        initializer
+    {
+        setPausable(_pausable);
+        setExpiryDuration(_expiryDuration);
+    }
+
+    function addToWhitelist(address _account)
+        external
+        onlyGovernorOrStrategist
+    {
+        whitelist[_account] = true;
+        emit Whitelisted(_account);
+    }
+
+    function removeFromWhitelist(address _account)
+        external
+        onlyGovernorOrStrategist
+    {
+        whitelist[_account] = false;
+        emit DeWhitelisted(_account);
+    }
+
+    /**
+     * @dev Pauses the managed contract for the duration of `expiryDuration`
+     * The pause action can be canceled by any user when the temp-pause expires.
+     * The contract MUST be in an UNPAUSED state.
+     * The function is one-time-use (The contract cannot be temp-paused by the same account twice).
+     */
+    function tempPause() external onlyWhitelistedOrGovernorOrStrategist {
```

Comment:
When a whitelisted user calls `pause()` to pause the contract, their role should be removed, blocking their ability to pause again. This prevents a single account from permanently locking up funds by calling unpause/pause over and over again, in the absence of action from the strategist.

This also removes the need for the pauseExpiry mapping variable.